server/sonar-web/src/main/js/apps/issues/components/App.js

```diff
@@ -652,8 +654,30 @@ export default class App extends React.PureComponent {
     });
   };
 
-  handleIssueCheck = (issue /*: string */) => {
+  handleIssueCheck = (issue /*: string */, event /*: Event */) => {
+    // Selecting multiple issues with shift+click
+    const { lastChecked } = this.state;
+    if (event.shiftKey && lastChecked !== null) {
+      const issueIndexes = this.state.issues.map(issue => issue.key);
+      const currentIssueIndex = issueIndexes.indexOf(issue);
+      const lastSelectedIndex = issueIndexes.indexOf(lastChecked);
+      const shouldCheck = this.state.checked.includes(lastChecked);
+      let { checked } = this.state;
+      if (currentIssueIndex < 0) {
+        return;
+      }
+      const start = Math.min(currentIssueIndex, lastSelectedIndex);
+      const end = Math.max(currentIssueIndex, lastSelectedIndex);
+      for (let i = start; i < end + 1; i++) {
+        checked = shouldCheck
+          ? union(checked, [this.state.issues[i].key])
+          : without(checked, this.state.issues[i].key);
+      }
+      this.setState(state => ({ checked }));
```

Comment:
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous